### PR TITLE
refactor: start_docker()

### PR DESCRIPTION
**Pull request by AkshayAwate**
Running a command and then checking its exit status with `$?` against 0 is redundant. 
More info can be found: https://www.shellcheck.net/wiki/SC2181
If this is good to go, then we can refactor other functions code too.
Thanks

### CODE REVIEW

File: deploy/install.sh [hunk 1]
```diff
@@ -204,9 +204,14 @@ start_docker() {
...
+        # if [[ -z $sudo_cmd ]]; then
+        #     docker ps > /dev/null && true
+        #     if [[ $? -ne 0 ]]; then
+        #         request_sudo
+        #     fi
+        # fi
         if [[ -z $sudo_cmd ]]; then
-            docker ps > /dev/null && true
-            if [[ $? -ne 0 ]]; then
+            if ! docker ps > /dev/null && true; then
```
Comment (prashant-shahi) [inline deploy/install.sh:214]: Good practice :+1:

**Review by prashant-shahi — CHANGES_REQUESTED** [review]
Thanks for the pull request!

Can you also address similar ones in line number 268, 272 and 290?

```
[Line 268:](javascript:setPosition(268, 15))
        if (( $EUID != 0 )); then
              ^-- [SC2004](https://www.shellcheck.net/wiki/SC2004) (style): $/${} is unnecessary on arithmetic variables.
 
[Line 272:](javascript:setPosition(272, 19))
            if [[ $? -ne 0 ]] && ! $sudo_cmd -v; then
                  ^-- [SC2181](https://www.shellcheck.net/wiki/SC2181) (style): Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.
 
[Line 290:](javascript:setPosition(290, 7))
if (( $EUID != 0 )); then
      ^-- [SC2004](https://www.shellcheck.net/wiki/SC2004) (style): $/${} is unnecessary on arithmetic variables.
 
[Line 307:](javascript:setPosition(307, 7))
if [[ $? -ne 0 ]]; then
      ^-- [SC2181](https://www.shellcheck.net/wiki/SC2181) (style): Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.
```

**Comment by AkshayAwate** [comment]
@prashant-shahi since there is no arithmatic calculation involved using `$EUID`, we can ignore refactoring 
[L268](https://github.com/SigNoz/signoz/blob/develop/deploy/install.sh#L268) and [L290](https://github.com/SigNoz/signoz/blob/develop/deploy/install.sh#L290)

**Comment by prashant-shahi** [comment]
@AkshayAwate `shellcheck` was asking to use `EUID` instead of `$EUID` for the arithmetic operation.
But anyways, we can address that and other best practices later.

**Pull request merged**